web/src/scheduled_messages.js

```diff
@@ -319,3 +323,40 @@ export function update_send_later_options() {
         $("#send_later_options").replaceWith(render_send_later_modal_options(filtered_send_opts));
     }
 }
+
+export function show_schedule_message_indicator() {
+    const scheduled_messages = Object.values(scheduled_messages_data);
+
+    const matching_scheduled_messages = scheduled_messages.filter((scheduled_message) => {
+        if (scheduled_message.failed === true) {
+            return false;
+        }
+        switch (scheduled_message.type) {
+            case "private":
+                if (scheduled_message.to.toString() === narrow_state.pm_ids_string()) {
+                    return true;
+                }
+                break;
+            case "stream":
+                if (
+                    scheduled_message.topic === narrow_state.topic() &&
+                    scheduled_message.to === narrow_state.stream_sub().stream_id
+                ) {
+                    return true;
+                }
+                break;
+        }
+        return false;
+    });
+
+    $("#scheduled_message_indicator").empty();
+    const scheduled_message_count = matching_scheduled_messages.length;
+    if (scheduled_message_count > 0) {
+        $("#scheduled_message_indicator").html(
+            render_scheduled_messages_indicator({
+                several_messages: scheduled_message_count > 1,
+                scheduled_message_count,
+            }),
+        );
+    }
```

Comment:
I think this logic is incorrect, in that a narrow can be `stream:foo topic:bar sender:quux`, and the logic here isn't really checking for that. If the product goal is to just show these in single-conversation views, which seems fine, `filter.is_conversation_view` is the check you want. 


We could also try doing something involving `filter.predicate`, but I'm not sure it would be easy to get working, as scheduled messages don't have the same format as actual messages.

@alya do we care about non-conversation views for this feature? They certainly seem like 5x less important to me.

Reply:
Added filter.is_conversation_view in latest commit.